### ISSUE TITLE
[BUG] CPSAM finetuning produces empty masks (3D)

### ISSUE THREAD

**Issue by postnubilaphoebus** [MouseLand/cellpose#1369]
Hi, I tried finetuning CPSAM on some greyscale images, but the result was an empty mask.
It is worth noting that I am using the 3d model of cellpose using pre-sliced 2d views for training.
Here is the logging during fine-tuning and inference:

```
2025-11-11 15:54:14,094 [INFO] WRITING LOG OUTPUT TO /home/USER/.cellpose/run.log
2025-11-11 15:54:14,094 [INFO] 
cellpose version:       3.0.7.dev3+g509ffca 
platform:               linux 
python version:         3.8.5 
torch version:          2.4.1+cu121
2025-11-11 15:54:14,098 [INFO] not all flows are present, running flow generation for all images
2025-11-11 15:54:14,138 [INFO] 30 / 30 images in /home/USER/Desktop/cloned/cellpose/cellpose/Mouse-Skull-Nuclei-CBG/train/redone folder have labels
2025-11-11 15:54:14,141 [INFO] not all flows are present, running flow generation for all images
2025-11-11 15:54:14,172 [INFO] 30 / 30 images in /home/USER/Desktop/cloned/cellpose/cellpose/Mouse-Skull-Nuclei-CBG/validation folder have labels
2025-11-11 15:54:14,454 [INFO] ** TORCH CUDA version installed and working. **
2025-11-11 15:54:14,454 [INFO] >>>> using GPU (CUDA)
2025-11-11 15:54:16,769 [INFO] >>>> loading model /home/USER/.cellpose/models/cpsam
2025-11-11 15:54:17,657 [INFO] computing flows for labels
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:02<00:00, 10.15it/s]
2025-11-11 15:54:20,623 [INFO] computing flows for labels
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:02<00:00, 12.16it/s]
2025-11-11 15:54:23,095 [INFO] >>> computing diameters
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:00<00:00, 1952.84it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:00<00:00, 3508.51it/s]
2025-11-11 15:54:23,120 [INFO] >>> normalizing {'lowhigh': None, 'percentile': None, 'normalize': True, 'norm3D': True, 'sharpen_radius': 0, 'smooth_radius': 0, 'tile_norm_blocksize': 0, 'tile_norm_smooth3D': 1, 'invert': False}
2025-11-11 15:54:27,322 [INFO] >>> n_epochs=100, n_train=30, n_test=30
2025-11-11 15:54:27,322 [INFO] >>> AdamW, learning_rate=0.00001, weight_decay=0.10000
2025-11-11 15:54:27,324 [INFO] >>> saving model to /home/USER/Desktop/cloned/cellpose/cellpose/models/mouse_skull
2025-11-11 15:54:57,221 [INFO] 0, train_loss=0.4272, test_loss=0.3815, LR=0.000000, time 29.90s
2025-11-11 15:57:13,057 [INFO] 5, train_loss=0.4354, test_loss=0.3462, LR=0.000006, time 165.73s
2025-11-11 15:59:35,113 [INFO] 10, train_loss=0.3594, test_loss=0.2976, LR=0.000010, time 307.79s
2025-11-11 16:04:00,276 [INFO] 20, train_loss=0.2767, test_loss=0.2726, LR=0.000010, time 572.95s
2025-11-11 16:08:22,929 [INFO] 30, train_loss=0.2513, test_loss=0.2668, LR=0.000010, time 835.61s
2025-11-11 16:12:46,905 [INFO] 40, train_loss=0.2382, test_loss=0.2657, LR=0.000010, time 1099.58s
2025-11-11 16:17:14,081 [INFO] 50, train_loss=0.2276, test_loss=0.2669, LR=0.000005, time 1366.76s
2025-11-11 16:21:44,146 [INFO] 60, train_loss=0.2305, test_loss=0.2671, LR=0.000001, time 1636.82s
2025-11-11 16:26:10,801 [INFO] 70, train_loss=0.2340, test_loss=0.2671, LR=0.000000, time 1903.48s
2025-11-11 16:30:38,541 [INFO] 80, train_loss=0.2195, test_loss=0.2673, LR=0.000000, time 2171.22s
2025-11-11 16:35:07,640 [INFO] 90, train_loss=0.2293, test_loss=0.2672, LR=0.000000, time 2440.32s
2025-11-11 16:39:06,287 [INFO] saving network parameters to /home/USER/Desktop/cloned/cellpose/cellpose/models/mouse_skull
2025-11-11 16:39:10,326 [INFO] running YX: 125 planes of size (512, 256)
2025-11-11 16:39:10,327 [INFO] 0%|          | 0/125 [00:00<?, ?it/s]
2025-11-11 16:39:41,937 [INFO] 15%|#5        | 19/125 [00:31<02:56,  1.66s/it]
2025-11-11 16:39:58,118 [INFO] 15%|#5        | 19/125 [00:47<02:56,  1.66s/it]
2025-11-11 16:40:11,999 [INFO] 30%|##9       | 37/125 [01:01<02:26,  1.67s/it]
2025-11-11 16:40:28,120 [INFO] 30%|##9       | 37/125 [01:17<02:26,  1.67s/it]
2025-11-11 16:40:43,486 [INFO] 45%|####4     | 56/125 [01:33<01:54,  1.66s/it]
2025-11-11 16:40:58,121 [INFO] 45%|####4     | 56/125 [01:47<01:54,  1.66s/it]
2025-11-11 16:41:14,881 [INFO] 60%|######    | 75/125 [02:04<01:22,  1.66s/it]
2025-11-11 16:41:28,122 [INFO] 60%|######    | 75/125 [02:17<01:22,  1.66s/it]
2025-11-11 16:41:46,106 [INFO] 75%|#######5  | 94/125 [02:35<00:51,  1.65s/it]
2025-11-11 16:41:58,123 [INFO] 75%|#######5  | 94/125 [02:47<00:51,  1.65s/it]
2025-11-11 16:42:17,116 [INFO] 90%|######### | 113/125 [03:06<00:19,  1.65s/it]
2025-11-11 16:42:28,124 [INFO] 90%|######### | 113/125 [03:17<00:19,  1.65s/it]
2025-11-11 16:42:36,753 [INFO] 100%|##########| 125/125 [03:26<00:00,  1.65s/it]
2025-11-11 16:42:36,996 [INFO] running ZY: 512 planes of size (125, 256)
2025-11-11 16:42:36,997 [INFO] 0%|          | 0/128 [00:00<?, ?it/s]
2025-11-11 16:43:08,192 [INFO] 11%|#         | 14/128 [00:31<04:14,  2.23s/it]
2025-11-11 16:43:28,126 [INFO] 11%|#         | 14/128 [00:51<04:14,  2.23s/it]
2025-11-11 16:43:39,397 [INFO] 22%|##1       | 28/128 [01:02<03:42,  2.23s/it]
2025-11-11 16:43:58,128 [INFO] 22%|##1       | 28/128 [01:21<03:42,  2.23s/it]
2025-11-11 16:44:10,105 [INFO] 33%|###2      | 42/128 [01:33<03:10,  2.21s/it]
2025-11-11 16:44:28,129 [INFO] 33%|###2      | 42/128 [01:51<03:10,  2.21s/it]
2025-11-11 16:44:40,510 [INFO] 44%|####3     | 56/128 [02:03<02:38,  2.20s/it]
2025-11-11 16:44:58,130 [INFO] 44%|####3     | 56/128 [02:21<02:38,  2.20s/it]
2025-11-11 16:45:11,873 [INFO] 55%|#####4    | 70/128 [02:34<02:08,  2.21s/it]
2025-11-11 16:45:28,131 [INFO] 55%|#####4    | 70/128 [02:51<02:08,  2.21s/it]
2025-11-11 16:45:43,296 [INFO] 66%|######5   | 84/128 [03:06<01:37,  2.22s/it]
2025-11-11 16:45:58,133 [INFO] 66%|######5   | 84/128 [03:21<01:37,  2.22s/it]
2025-11-11 16:46:13,433 [INFO] 77%|#######6  | 98/128 [03:36<01:06,  2.20s/it]
2025-11-11 16:46:28,134 [INFO] 77%|#######6  | 98/128 [03:51<01:06,  2.20s/it]
2025-11-11 16:46:44,495 [INFO] 88%|########7 | 112/128 [04:07<00:35,  2.21s/it]
2025-11-11 16:46:58,135 [INFO] 88%|########7 | 112/128 [04:21<00:35,  2.21s/it]
2025-11-11 16:47:15,453 [INFO] 98%|#########8| 126/128 [04:38<00:04,  2.21s/it]
2025-11-11 16:47:19,769 [INFO] 100%|##########| 128/128 [04:42<00:00,  2.21s/it]
2025-11-11 16:47:19,910 [INFO] running ZX: 256 planes of size (125, 512)
2025-11-11 16:47:19,910 [INFO] 0%|          | 0/128 [00:00<?, ?it/s]
2025-11-11 16:47:51,112 [INFO] 15%|#4        | 19/128 [00:31<02:59,  1.64s/it]
2025-11-11 16:48:08,137 [INFO] 15%|#4        | 19/128 [00:48<02:59,  1.64s/it]
2025-11-11 16:48:21,486 [INFO] 29%|##8       | 37/128 [01:01<02:31,  1.67s/it]
2025-11-11 16:48:38,138 [INFO] 29%|##8       | 37/128 [01:18<02:31,  1.67s/it]
2025-11-11 16:48:53,222 [INFO] 44%|####3     | 56/128 [01:33<02:00,  1.67s/it]
2025-11-11 16:49:08,140 [INFO] 44%|####3     | 56/128 [01:48<02:00,  1.67s/it]
2025-11-11 16:49:24,842 [INFO] 59%|#####8    | 75/128 [02:04<01:28,  1.67s/it]
2025-11-11 16:49:38,141 [INFO] 59%|#####8    | 75/128 [02:18<01:28,  1.67s/it]
2025-11-11 16:49:56,427 [INFO] 73%|#######3  | 94/128 [02:36<00:56,  1.67s/it]
2025-11-11 16:50:08,142 [INFO] 73%|#######3  | 94/128 [02:48<00:56,  1.67s/it]
2025-11-11 16:50:26,649 [INFO] 88%|########7 | 112/128 [03:06<00:26,  1.67s/it]
2025-11-11 16:50:38,143 [INFO] 88%|########7 | 112/128 [03:18<00:26,  1.67s/it]
2025-11-11 16:50:52,523 [INFO] 100%|##########| 128/128 [03:32<00:00,  1.66s/it]
2025-11-11 16:50:52,762 [INFO] network run in 702.44s
2025-11-11 16:50:53,745 [WARNING] no seeds found in get_masks_torch - no masks found.
2025-11-11 16:50:53,913 [INFO] masks created in 0.55s

```

I already followed the general procedure of slicing my 3d volume into 2d slices using the make_train.py function, followed by regenerating the label indices between 1 and the maximum number of objects in that image.

This is my fine-tuning code (I renamed io to cio to avoid conflicts with internal libs):

```
import cio, models, train
import skimage
import torch
cio.logger_setup()
import numpy as np

train_dir = "/path/to/train"
test_dir = "path/to/val"

output = cio.load_train_test_data(train_dir, test_dir, image_filter="_img",
                                mask_filter="_masks", look_one_level_down=False)
images, labels, image_names, test_images, test_labels, image_names_test = output

model = models.CellposeModel(gpu=True, pretrained_model="cpsam")

testing_image = [skimage.io.imread("path/to/testing_image")]

testing_image = [np.expand_dims(testing_image[i], axis = 0).repeat(3).reshape(3, 125, 512, 256) for i in range(len(testing_image))]

model_path, train_losses, test_losses = train.train_seg(model.net,
                            train_data=images, train_labels=labels,
                            test_data=test_images, test_labels=test_labels,
                            weight_decay=0.1, learning_rate=1e-5,
                            n_epochs=100, model_name="mouse_skull")

masks, flows, styles = model.eval(x = testing_image, do_3D = True, channel_axis = 0, z_axis = 1)
```
